File-Formats/Presentation/Working-with-find-and-replace.md

```diff
@@ -0,0 +1,932 @@
+//Creates a storage file from FileSavePicker
+StorageFile storageFile = await savePicker.PickSaveFileAsync();
+//Saves changes to the specified storage file
+await pptxDoc.SaveAsync(storageFile);
+{% endhighlight %}
+
+{% highlight c# tabtitle="ASP.NET Core" %}
+ //Opens an existing presentation.           
+FileStream inputStream = new FileStream("Sample.pptx", FileMode.Open);
+using (IPresentation pptxDoc = Presentation.Open(inputStream))
+{
+	ISlide slide = pptxDoc.Slides[0];
+	//Finds all the occurrences of a particular text in the specific slide
+	ITextSelection[] textSelections = slide.FindAll("product", false, false);
+	foreach (ITextSelection textSelection in textSelections)
+	{
+		//Gets the found text as single text part
+		ITextPart textPart = textSelection.GetAsOneTextPart();
+		//Replace the text
+		textPart.Text = "Service";
+	}
+	//Save the PowerPoint Presentation as stream
+	FileStream outputStream = new FileStream("Output.pptx", FileMode.Create);
+	pptxDoc.Save(outputStream);
+	//Release all resources of the stream
+	outputStream.Dispose();
+	//Closes the Presentation instance
+	pptxDoc.Close();
+}
+{% endhighlight %}
+
+{% highlight c# tabtitle="Xamarin" %}
+//"App" is the class of Portable project.
+Assembly assembly = typeof(App).GetTypeInfo().Assembly;
+Stream inputStream = assembly.GetManifestResourceStream("SampleBrowser.Presentation.Samples.Template.Sample.pptx");
+//Loads or open an PowerPoint Presentation
+IPresentation pptxDoc = Presentation.Open(inputStream);
+ISlide slide = pptxDoc.Slides[0];
+//Finds all the occurrences of a particular text in the specific slide
+ITextSelection[] textSelections = slide.FindAll("product", false, false);
+foreach (ITextSelection textSelection in textSelections)
+{
+	//Gets the found text as single text part
+	ITextPart textPart = textSelection.GetAsOneTextPart();
+	//Replace the text
+	textPart.Text = "Service";
+}
+//Create new memory stream to save Presentation.
+MemoryStream stream = new MemoryStream();
+//Save Presentation in stream format.
+pptxDoc.Save(stream);
+//Close the presentation
+pptxDoc.Close();
+stream.Position = 0;
+//The operation in Save under Xamarin varies between Windows Phone, Android and iOS platforms. Please refer presentation/xamarin section for respective code samples.
+if (Device.OS == TargetPlatform.WinPhone || Device.OS == TargetPlatform.Windows)
+	Xamarin.Forms.DependencyService.Get<ISaveWindowsPhone>().Save("Output.pptx", "application/vnd.openxmlformats-officedocument.presentationml.presentation", stream);
+else
+	Xamarin.Forms.DependencyService.Get<ISave>().Save("Output.pptx", "application/vnd.openxmlformats-officedocument.presentationml.presentation", stream);
+{% endhighlight %}
+
+{% endtabs %}  
+
+## Find and highlight in PowerPoint presentation
+
+.NET PowerPoint library allows you to find all the occurences of a text in the PowerPoint presentation and highlight it.
```

Comment:
.NET PowerPoint library allows you to find all the occurrences of a text in the PowerPoint presentation and highlight it.

Reply:
Modified as you said.